Add documentation for JEP-235 (#4789)
jenkins-infra/jenkins.io: content/doc/book/security/controller-isolation/agent-to-controller.adoc

:hide-uri-scheme:
endif::[]

This section describes how to customize the rules restricting agents' access to the Jenkins controller in Jenkins 2.325 and earlier.

// TODO Also mention first LTS once it's known
NOTE: In Jenkins 2.326, the ability to disable or customize the agent-to-controller security system has been removed without replacement.

> This documentation only applies to older releases of Jenkins.

Reviewer: ```suggestion
This documentation applies only to older releases of Jenkins.
```

For documentation specific to the change in Jenkins 2.326, see link:../jep-235[Agent &rarr; Controller Security Changes in 2.326].

> For advanced users who may wish to allow certain access patterns from the agents to the Jenkins controller, Jenkins allows administrators to create specific exemptions from the built-in access control rules.

Reviewer: ```suggestion
Sophisticated administrators can use the built-in access control rules to create specific exemptions for certain access patterns from the agents to the Jenkins controller.
```

> This should only be necessary when trying to run plugins that haven't been adapted yet to be compatible.

Reviewer: ```suggestion
This should only be necessary when trying to run plugins that have not yet been adapted to be compatible.
```
